applications/luci-app-filemanager/htdocs/luci-static/resources/view/system/filemanager.js

```diff
@@ -0,0 +1,2288 @@
+            }
+        }
+        numeric += subtotal.toString();
+    }
+
+    if (specialPerms > 0) {
+        numeric = specialPerms.toString() + numeric;
+    }
+
+    return numeric;
+}
+
+// Function to get file list using fs.exec()
+function getFileList(path) {
+    return fs.exec('/bin/ls', ['-lA', '--full-time', path]).then(function(res) {
+        if (res.code !== 0) {
+            var errorMessage = res.stderr ? res.stderr.trim() : 'Unknown error';
+            return Promise.reject(new Error('Failed to list directory: ' + errorMessage));
+        }
+
+        // Check if  res.stdout id defined and set null string as a default
+        var stdout = res.stdout || '';
+        var lines = stdout.trim().split('\n');
+        var files = [];
+
+        // If folder is empty, lines would contain an empty array and we simply return an empty file list
+        lines.forEach(function(line) {
+            if (line.startsWith('total') || !line.trim()) return;
+
+            var parts = line.match(/^([\-dl])[rwx\-]{2}[rwx\-Ss]{1}[rwx\-]{2}[rwx\-Ss]{1}[rwx\-]{2}[rwx\-Tt]{1}\s+\d+\s+(\S+)\s+(\S+)\s+(\d+)\s+([\d\-]+\s+[\d\:\.]{8,12}\s+\+\d{4})\s+(.+)$/);
+            if (!parts || parts.length < 7) {
+                console.warn('Failed to parse line:', line);
+                return;
+            }
+            var typeChar = parts[1];
+            var permissions = line.substring(0, 10);
+            var owner = parts[2];
+            var group = parts[3];
+            var size = parseInt(parts[4], 10);
+            var dateStr = parts[5];
+            var name = parts[6];
+
+            var type = '';
+            var target = null;
+            if (typeChar === 'd') {
+                type = 'directory';
+            } else if (typeChar === '-') {
+                type = 'file';
+            } else if (typeChar === 'l') {
+                type = 'symlink';
+                var linkParts = name.split(' -> ');
+                name = linkParts[0];
+                target = linkParts[1] || '';
+            } else {
+                type = 'unknown';
+            }
+
+            var mtime = Date.parse(dateStr);
+
+            if (type === 'symlink' && target && size === 4096) {
+                size = -1;
+            }
+
+            files.push({
+                name: name,
+                type: type,
+                size: size,
+                mtime: mtime / 1000,
+                owner: owner,
+                group: group,
+                permissions: permissions.substring(1),
+                numericPermissions: symbolicToNumeric(permissions.substring(1)),
+                target: target
+            });
+        });
+        return files;
+    });
+}
+
+function insertCss(cssContent) {
+    var styleElement = document.createElement('style');
+    styleElement.type = 'text/css';
+    styleElement.appendChild(document.createTextNode(cssContent));
+    document.head.appendChild(styleElement);
+}
+
+var cssContent = '' +
```

Comment:
maybe rework this as a ` quoted string with {variables}

Reply:
Done